tests/subsys/logging/log_syst/src/main.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Test log message
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include "mock_backend.h"
+#include <sys/printk.h>
+#include <logging/log_backend.h>
+#include <logging/log_backend_std.h>
+#include <logging/log_ctrl.h>
+#include <logging/log_output.h>
+#include <tc_util.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <stdlib.h>
+
+#define TEST_MSG_0 "0 args"
+
+MOCK_LOG_BACKEND_DEFINE(backend1, false);
+
+#define LOG_MODULE_NAME test
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+/**
+ * @brief Testcase to validate that the mock backend would use the expected
+ * processing function from the function pointer table format_table
+ */
+void test_log_syst_format_table_selection(void)
+{
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+	uint32_t test_log_type_syst = LOG_OUTPUT_SYST;
+
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_syst);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_syst_process,
+					"Correct Function pointer for SYST log\n"
+					"log format was not selected %x vs %x",
```

Comment:
Why not `%p` to show the pointer value?

Reply:
yeahh %p is probably better to use here. Changed to %p. Thanks!!